File: server/models/Product.js
// Phase 5: MongoDB Product Schema Specification
export const ProductSchema = {
  name: { type: String, required: true },
  category: { type: String, required: true, index: true },
  price: { type: Number, required: true },
  originalPrice: { type: Number },
  discount: { type: Number, default: 0 },
  rating: { type: Number, default: 5.0 },
  reviewCount: { type: Number, default: 0 },
  stock: { type: Number, required: true, default: 0 },
  featured: { type: Boolean, default: false },
  bestSeller: { type: Boolean, default: false },
  isNew: { type: Boolean, default: true },
  dealOfTheDay: { type: Boolean, default: false },
  images: [{ type: String }],
  description: { type: String },
  features: [{ type: String }],
  colors: [{
    name: String,
    hex: String,
    inStock: Boolean
  }],
  sizes: [{ type: String }],
  specs: { type: Map, of: String },
  createdAt: { type: Date, default: Date.now }
};

File: backend/app/models/product.py
"""Product models for CartVerse"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class ProductBase(BaseModel):
    """Base product model"""
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    price: float = Field(..., gt=0)
    category: str
    stock: int = Field(default=0, ge=0)
    rating: float = Field(default=0, ge=0, le=5)
    discount: float = Field(default=0, ge=0, le=100)
    image_url: Optional[str] = None
    featured: bool = False
    best_seller: bool = False
    is_new: bool = False
    deal_of_the_day: bool = False

class ProductCreate(ProductBase):
    """Create product request"""
    pass

class ProductUpdate(BaseModel):
    """Update product request"""
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    category: Optional[str] = None
    stock: Optional[int] = None
    rating: Optional[float] = None
    discount: Optional[float] = None
    image_url: Optional[str] = None
    featured: Optional[bool] = None
    best_seller: Optional[bool] = None
    is_new: Optional[bool] = None
    deal_of_the_day: Optional[bool] = None

class Product(ProductBase):
    """Product response model"""
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ProductPage(BaseModel):
    """Paginated products response"""
    items: List[Product]
    total: int
    page: int
    page_size: int
    has_more: bool
